cookbook/securing-your-application.rst

```diff
@@ -0,0 +1,124 @@
+Securing your application
+=========================
+
+Sulu is delivered with two different possiblities to protect parts of your
+application. The first are the permissions based on security contexts, which
+allow you to restrict access to entire parts of your application or Sulu. The
+permissions for this kind of security are managed on a roles level. In addition
+to that the localization for which these permissions are valid has to be
+defined on the assignment of the role to the user.
+
+The second way is to protect the access on a per-object basis. These
+permissions are set on the specific object. The user still has to have the
+correct localizations assigned in order to gain access.
+
+This tutorial will show how to use Sulu's security functionality on your own
+application specific code.
+
+Protect content using a security context
+----------------------------------------
+
+This chapter describes how to protect an entire part of your application (but
+not a specific object).
+
+Define your security context
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First of all you have to define the security context, which is represented by a
+simple string. This is done in the ``Admin`` class of your Bundle:
+
+.. code-block:: php
+    
+    <?php
+
+    namespace Acme\Bundle\ExampleBundle\Admin;
+
+    use Sulu\Bundle\AdminBundle\Admin\Admin;
+
+    class AcmeExampleAdmin extends Admin
+    {
+        // ...
+
+        public function getSecurityContexts()
+        {
+            return array(
+                'Sulu' => array(
+                    'Acme' => array(
+                        'sulu.acme.example'
+                    )
+                );
+            );
+        }
+
+        // ...
+    }
+
+This information is defined in the ``getSecurityContexts`` method, which should
+return an array. The first level describes the used system being either Sulu
```

Comment:
the system to which the security context applies - this would either be Sulu (for stuff in the administration) or a different context that you have defined manually.